extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/runtime/MongoOperations.java

```diff
@@ -0,0 +1,547 @@
+package io.quarkus.mongodb.panache.runtime;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.bson.BsonDocument;
+import org.bson.BsonDocumentWriter;
+import org.bson.BsonValue;
+import org.bson.Document;
+import org.bson.codecs.Codec;
+import org.bson.codecs.EncoderContext;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.logging.Logger;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.InsertOneModel;
+import com.mongodb.client.model.ReplaceOneModel;
+import com.mongodb.client.model.ReplaceOptions;
+import com.mongodb.client.model.UpdateOptions;
+import com.mongodb.client.model.WriteModel;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.mongodb.panache.MongoEntity;
+import io.quarkus.mongodb.panache.PanacheQuery;
+import io.quarkus.panache.common.Parameters;
+import io.quarkus.panache.common.Sort;
+
+public class MongoOperations {
+    private static final Logger LOGGER = Logger.getLogger(MongoOperations.class);
+    public static final String ID = "_id";
+    public static final String MONGODB_DATABASE = "quarkus.mongodb.database";
+    //
+    // Instance methods
+
+    public static void insert(Object entity) {
+        MongoCollection collection = getMongoCollection(entity);
+        insert(collection, entity);
+    }
+
+    public static void insert(Iterable<?> entities) {
+        // not all iterables are re-traversal, so we traverse it once for copying inside a list
+        List<Object> objects = new ArrayList<>();
+        for (Object entity : entities) {
+            objects.add(entity);
+        }
+
+        if (objects.size() > 0) {
+            //get the first to get the collection from it
```

Comment:
`get the first document/entity and then retrieve its collection`?

Reply:
changed to `// get the first entity to be able to retrieve the collection with it`

Reply:
done